### PR TITLE
Space Travelers Hub V1.0

### PR DESCRIPTION
**Pull request by AlexHamn**
### Setup: 
- Create app and install basic dependencies:
  - Create React App.
  - Install [React Redux](https://react-redux.js.org/), [Redux Logger](https://www.npmjs.com/package/redux-logger) and [React Router](https://reactrouter.com/web/guides/quick-start).

- Create website header with empty nav:
  - Download the [free image](https://www.flaticon.com/free-icon/planet_3212567?term=space&page=1&position=19&page=1&position=19&related_id=3212567&origin=style) for the app logo.
  - Create **empty** `navigation`.

- Create **empty** `My profile` tab:
  - Create a route and a view component. Use `<NavLink />` for the page navigation links and style active class to indicate which section/page the user is currently on (underline active navigation link).

- Create the basic structure for Rockets:
  - Create a route and a view component. Use `<NavLink />` for the page navigation links and style active class to indicate which section/page the user is currently on (underline active navigation link).
  - Create a directory for all Redux state slice files.

- Create the basic structure for Missions:
  - Create a route and a view component. Use `<NavLink />` for the page navigation links and style active class to indicate which section/page the user is currently on (underline active navigation link).
  - Create a directory for all Redux state slice files.

### Fetch data:
- Fetch rockets data:
    - When the application starts, fetch data from the Rockets endpoint  (https://api.spacexdata.com/v3/rockets) (as Rockets is the default view).

  - Once the data gets fetched, dispatch an action to store the selected data in Redux store:

    - id
    - name
    - type
    - flickr_images

  - Only dispatch those actions once and do not add data to store on every re-render (i.e. when changing views / using navigation).

- Fetch missions data:
  - Fetch data from the Missions endpoint  (https://api.spacexdata.com/v3/missions) when a user navigates to the Missions section.

  - Once the data gets fetched, dispatch an action to store the selected data in Redux store:

    - mission_id
    - mission_name
    - description

  - Only dispatch those actions once and do not add data to store on every re-render (i.e. when changing views / using navigation).

### Lists render:
- Render rockets list:
  - Use `useSelector()` Redux Hook to select the state slices and render lists of rockets in corresponding routes.
  - Style the whole application using [React Bootstrap](https://react-bootstrap.github.io/).
  - Render a list of rockets (as per design). For the image of a rocket, use the first image in the array of `flickr_images`.
 
- Render missions list:
  - Use `useSelector()` Redux Hook to select the state slices and render lists of rockets in corresponding routes.
  - Style the whole application using [React Bootstrap](https://react-bootstrap.github.io/).
  - Render a table with the missions' data (as per design).

### Actions: 
- Implement rocket booking:
  - When a user clicks the "Reserve rocket" button, an action gets dispatched to update the store. Get the ID of the reserved rocket and update the state. 
- Place all logic in the reducer. In the React view file, only dispatch the action with the correct rocket ID as an argument. 

- Implement mission joining:
  - When a user clicks the "Join Mission" button, an action gets dispatched to update the store. Get the ID of the selected mission and update the state.
- Place all logic in the reducer. In the React view file, only dispatch the action with the correct mission ID as an argument.

- Implement rocket booking cancellation:
  - Follow the same logic as the "Reserve rocket" - but set the `reserved` key to `false`. 
  - Dispatch these actions upon clicking on the corresponding buttons.

- Implement mission leaving:
  - Follow the same logic as the "Join mission" - but set the `reserved` key to `false`. 
  - Dispatch these actions upon clicking on the corresponding buttons. 

### Conditional components:
- Switch badges for Rockets:
  - Rockets that have already been reserved show a "Reserved" badge and "Cancel reservation" button instead of the default "Reserve rocket" (as per design).

- Switch badges for Missions:
  - Missions that the user has joined already show a badge "Active Member" instead of the default "NOT A MEMBER" and a button "Leave Mission" instead of the "Join Mission" button (as per design).

### Filters:
- Display reserved rockets in My profile:
  - Render a list of all reserved rockets (use `filter()`) on the "My profile" page.

- Display joined missions in My profile:
  - Render a list of all joined missions (use `filter()`) on the "My profile" page.

### CODE REVIEW

File: src/setupTests.js [hunk 1]
```diff
@@ -1,5 +0,0 @@
-// jest-dom adds custom jest matchers for asserting on DOM nodes.
-// allows you to do things like:
-// expect(element).toHaveTextContent(/react/i)
-// learn more: https://github.com/testing-library/jest-dom
```
Comment (Mihndim2020) [inline src/setupTests.js:4]: - [ ] As it is a requirement for this project under technical set up, and bests practice to ensure your app is not broken by new changes, please kindly add tests. See screenshots below and kindly make appropriate changes. 

test

test2

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file
:heavy_check_mark: The project is running as expected

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file
:heavy_check_mark: The project is running as expected

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments above :point_up_2:  review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by er-contreras — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

- I see that you did the work, well done!

Felix

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**